src/components/UserManagement/UserOrderHistory/UserOrderHistory.tsx

```diff
@@ -321,10 +328,26 @@ export default class UserOrderHistory extends React.Component<Props, any> {
   ];
   canLoadMore: boolean = true;
   modalRef: any;
+  static contextType = AppContext;
+  context!: AppContextState;
 
   constructor(props: Props) {
     super(props);
     this.modalRef = React.createRef();
+    this.state = {
+      orders: [],
+    };
+  }
+
+  componentDidMount() {
+    this.context.getOrders().then((orders) => {
```

Comment:
I think we can get 100 orders or so and sort them here so that the most recent order is on top of the list.